openedx/core/djangoapps/api_admin/api/v1/views.py

```diff
@@ -0,0 +1,52 @@
+"""
+API Views.
+"""
+
+from django.contrib.auth.models import User
+from django.utils.translation import ugettext_lazy as _
+
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.generics import ListAPIView
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from openedx.core.djangoapps.api_admin.api.v1 import serializers as api_access_serializers
+from openedx.core.djangoapps.api_admin.models import ApiAccessRequest
+
+
+class ApiAccessRequestView(ListAPIView):
+    """
+    List Api Access Requests.
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated, )
+    serializer_class = api_access_serializers.ApiAccessRequestSerializer
+
+    def get_queryset(self):
+        username = self.request.query_params.get('username', None)
+        queryset = ApiAccessRequest.objects.all()
+
+        # Filter based on given username, if requesting user has staff access.
+        if self.request.user.is_staff:
+            if username:
+                try:
+                    user = User.objects.get(username=username)
+                except User.DoesNotExist:
+                    raise NotFound(_('No user with the username [{username}] exists.').format(username=username))
+
+                queryset = queryset.filter(user=user)
+
+        # Raise PermissionDenied error if requesting user is does not have access and is asking for
+        # filtering based on username.
+        elif username:
+            raise PermissionDenied(
+                _('Only staff users are permitted to filter by username. Remove the username parameter.')
+            )
+        # If requesting user does not have staff access then return
+        # only the instances created by the requesting user.
+        else:
+            queryset = queryset.filter(user=self.request.user)
```

Comment:
I think we should make it as default filter as we will be returning the normal users only their records.

Reply:
we discussed this in person, doing that will not get us desired results.

Reply:
I am OK with this change as we would have to provide others user's record to staff.